### PR TITLE
pc/storage/kdtree: range search

### PR DESCRIPTION
**Pull request by f-fl0**
This PR adds method to do a range search given a point i.e. retrieve all points within a specific range from that point.

### CODE REVIEW

File: pc/storage/search.go [hunk 1]
```diff
@@ -8,4 +8,5 @@ import (
 type Search interface {
 	pc.Vec3RandomAccessor
 	Nearest(p mat.Vec3, maxRange float32) (int, float32)
+	Range(p mat.Vec3, maxRange float32) ([]int, []float32)
```
[the remark this answers is not in the document]
Comment (f-fl0) [reply pc/storage/search.go:11]: Sure. Should we change `Nearest(p mat.Vec3, maxRange float32) (int, float32)` to `Nearest(p mat.Vec3, maxRange float32) Neighbor` as well?

[the remark this answers is not in the document]
Comment (f-fl0) [reply pc/storage/search.go]: Done in cdf39c32cda38cbcb0d67e06b24fc23b1d135882

File: pc/storage/kdtree/kdtree.go [hunk 2]
```diff
@@ -58,24 +59,84 @@ func (k *KDTree) Nearest(p mat.Vec3, maxRange float32) (int, float32) {
...
+			continue
+		}
+
+		nodesOrig := k.newNodeArray(nextNode)
+		ns := k.searchLeafNode(p, nodesOrig)
+		neighbor2 := k.nearestImpl(p, ns, neighbor1.DistSq)
+		k.poolNodeArray.Put(nodesOrig)
+		if neighbor2.ID >= 0 {
+			neighbor1.ID = neighbor2.ID
+			neighbor1.DistSq = neighbor2.DistSq
```
Comment (at-wat) [inline pc/storage/kdtree/kdtree.go:101]:
```suggestion
			neighbor1 = neighbor2
```

File: pc/storage/kdtree/kdtree.go [hunk 3]
```diff
@@ -306,3 +363,19 @@ func (s *indiceSorter) Less(i, j int) bool {
...
+	return len(ns.neighbors)
+}
+
+func (ns *neighborSorter) Swap(i, j int) {
+	ns.neighbors[i], ns.neighbors[j] = ns.neighbors[j], ns.neighbors[i]
+}
+
+func (ns *neighborSorter) Less(i, j int) bool {
+	return ns.neighbors[i].DistSq < ns.neighbors[j].DistSq
+}
```
Comment (at-wat) [inline pc/storage/kdtree/kdtree.go:381]:
```suggestion
type neighborSorter []storage.Neighbor

func (ns neighborSorter) Len() int {
	return len(ns)
}

func (ns neighborSorter) Swap(i, j int) {
	ns[i], ns[j] = ns[j], ns[i]
}

func (ns neighborSorter) Less(i, j int) bool {
	return ns[i].DistSq < ns[j].DistSq
}
```
and
```diff
-       sort.Sort(&neighborSorter{neighbors: neighbors})
+       sort.Sort(neighborSorter(neighbors))
```

Comment (f-fl0) [reply pc/storage/kdtree/kdtree.go]: done in 053d68b2de03277c23c61fadb0a16367496dcc2a

**Pull request merged**